subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -1407,7 +1524,14 @@ int bt_bap_broadcast_assistant_rem_src(struct bt_conn *conn, uint8_t src_id)
 		LOG_DBG("handle not set");
 
 		return -EINVAL;
-	} else if (inst->busy) {
+	} else if (atomic_test_bit(inst->flags, BAP_BA_FLAG_DISCOVER_IN_PROGRESS) ||
+		   atomic_test_bit(inst->flags, BAP_BA_FLAG_READ_IN_PROGRESS) ||
+		   atomic_test_and_set_bit(inst->flags, BAP_BA_FLAG_WRITE_IN_PROGRESS)) {
```

Comment:
There is technically a small chance that if this function is called in Thread A and another API function is called in Thread B, that when we test `BAP_BA_FLAG_WRITE_IN_PROGRESS` here in Thread A, `BAP_BA_FLAG_DISCOVER_IN_PROGRESS` may have been set in thread B. 

How can do we prevent this (rare) issue? 
@jhedberg any ideas?

Reply:
> What's preventing this from return `false` when it should be `true` if the bit has changed between `atomic_val_t val = atomic_get(ATOMIC_ELEM(target, bit));` and `return (1 & (val >> (bit & (ATOMIC_BITS - 1)))) != 0;`?

As per our discussion on Discord, nothing is preventing this, but changing the implementation to be "more atomic" wouldn't give you anything, since then you just move the "risky place" (i.e. where you could be preempted) to right before or right after the function call. If you need to expand the critical section then you need some separate locking primitive.